tft.py

```diff
@@ -532,114 +468,122 @@ def check_if_gold_at_least(num: int) -> bool:
     return True
 
 
+def determine_minimum_round() -> int:
+    """
+    Determines minimum round we are at.
+    Prioritizes PvE markers, falls back to the round display.
+
+    Returns:
+        The major round as an integer.
+
+    """
+    if (
+        onscreen(CONSTANTS["game"]["round"]["krugs_inactive"], 0.9)
+        or onscreen(CONSTANTS["game"]["round"]["krugs_active"], 0.9)
+    ):
+        return 2
+
+    if (
+        onscreen(CONSTANTS["game"]["round"]["wolves_inactive"], 0.9)
+        or onscreen(CONSTANTS["game"]["round"]["wolves_active"], 0.9)
+    ):
+        return 3
+
+    if (
+        onscreen(CONSTANTS["game"]["round"]["threat_inactive"], 0.9)
+        or onscreen(CONSTANTS["game"]["round"]["threat_active"], 0.9)
+    ):
+        return 4
+
+    if onscreen(CONSTANTS["game"]["round"]["1-"]):
+        return 1
+
+    for i in range(1, 7):
+        if onscreen(CONSTANTS["game"]["round"][f"{i}-"]):
+            return i
+
+    logger.debug("Could not determine minimum round, returning 0.")
+    return 0
+
+
 def main_game_loop() -> None:  # pylint: disable=too-many-branches
     """The main in-game loop.
 
     Skips 5 second increments if a pause logic request is made, repeating until toggled or an event triggers an early exit.
 
-    Support for forfeiting early does exist but is rarely tested as the main author does not use this feature.
+    Support for forfeiting early does exist.
     """
     should_exit = False
     while should_exit is False:

```

Comment:
I wonder if, since L524 is doing a `break` anyways, this makes more sense to just omit the extra variable 🤔 
```suggestion
    while True:

```

Reply:
Did you one better and added an additional fail condition. Noticed that if going down for maintenance (which happened today) the game continues to try to reconnect? Not sure why, but it should be taken care of now.

Reply:
Ahh totally fair! I think previously I had it doing 5 minute delays (or similar) when it detected a 'maintenance' error screen, so that it would only attempt a few times during a 20 minute period instead of constantly 😅 (if I didn't, I meant to add that)